### DISCUSSION TITLE
Could not find a version for "xxx" in the root "package.json" when collecting shared packages for the Module Federation setup. The package will not be shared.

### DISCUSSION THREAD

**Discussion by sonikasharma1403** [nrwl/nx#15350] [General]
We are sharing a couple of libraries with NX module federation, but during the build process we are getting warnings - "Could not find a version for "xxx" in the root "package.json" when collecting shared packages for the Module Federation setup. The package will not be shared.". 

The library is coming up from the shared package dependency and it should not be shared.

My doubt here is, why should we bloat our main `package.json` just to avoid this warning?

**Comment by sonikasharma1403** [comment]
For instance, We have marked `some_library` as shared. Which has a dependency on `lodash` which is not part of the root package JSON. Now when we build the angular application, it gives us a warning 

"Could not find a version for "lodash" in the root "package.json" when collecting shared packages for the Module Federation setup. The package will not be shared."

We don't want to share the package, as it will cause another HTTP call just to get this chunk. Since, we are on HTTP 1, it will become very slow. 

Is there any way we can achieve this without bloating the root `package.json`

**Comment by timothylombrana** [comment]
I'm having the same issue, did you ever solve the warning @sonikasharma1403 ?

**Comment by diandsonc** [comment]
@sonikasharma1403 To eliminate this warning, you can either install the packages in your project or configure to ignore them using the following approach:

```typescript
const config: ModuleFederationConfig = {
  shared: (libraryName, sharedConfig) => {
    if (['buffer', 'events', 'string_decoder', '@nx/webpack'].includes(libraryName)) {
      return { singleton: true };
    }

    // Add your packages here
    if (['lodash'].includes(libraryName)) {
      return false;
    }

    return sharedConfig;
  },
};

export default config;
